src/webgl/loading.js

```diff
@@ -158,6 +158,57 @@ p5.prototype.loadModel = function(path,options) {
   model.gid = `${path}|${normalize}`;
   const self = this;
 
+  async function getMaterials(lines){
+    const parsedMaterialPromises=[];
+    const mtlPaths=[];
+    return new Promise(async (resolve,reject)=>{
+      for (let i = 0; i < lines.length; i++) {
+        const mtllibMatch = lines[i].match(/^mtllib (.+)/);
+        if (mtllibMatch) {
+          let mtlPath='';
+          const mtlFilename = mtllibMatch[1];
+          const objPathParts = path.split('/');
+          if(objPathParts.length > 1){
+            objPathParts.pop();
+            const objFolderPath = objPathParts.join('/');
+            mtlPath = objFolderPath + '/' + mtlFilename;
+          }else{
+            mtlPath = mtlFilename;
+          }
+          try {
+            if(await fileExists(mtlPath)){
+              const parsedMaterialsIndividual = await parseMtl(self, mtlPath);
```

Comment:
By adding `await` here, we're waiting for each material to load one at a time rather than waiting for all of them to load in parallel at the `Promise.all` below on line 195. Does this still work if we take out the `await` on this line, given that we already wait for all of them later?

Reply:
Removing `await` here would result in parallel loading of all mtl filled and then `await` with `promise.all` which is more efficient in handling all files together , better than sequential parsing of each mtl file.